OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs

```diff
@@ -482,19 +520,15 @@ void CloseWindow()
 			var techLevel = optionsBin.GetOrNull<DropDownButtonWidget>("TECHLEVEL_DROPDOWNBUTTON");
 			if (techLevel != null)
 			{
-				var techTraits = modRules.Actors["player"].TraitInfos<ProvidesTechPrerequisiteInfo>().ToList();
-				techLevel.IsVisible = () => techTraits.Count > 0;
-
-				var techLevelDescription = optionsBin.GetOrNull<LabelWidget>("TECHLEVEL_DESC");
-				if (techLevelDescription != null)
-					techLevelDescription.IsVisible = () => techTraits.Count > 0;
+				var mapOptions = new CachedTransform<Map, MapOptionsInfo>(
+					map => map.Rules.Actors["world"].TraitInfo<MapOptionsInfo>());
 
-				techLevel.IsDisabled = () => configurationDisabled() || Map.Options.TechLevel != null || techTraits.Count <= 1;
+				techLevel.IsDisabled = () => configurationDisabled() || mapOptions.Update(Map).TechLevelLocked;
```

Comment:
This doesn't disable the dropdown when there are no techlevels defined (as it is the case in TS atm). Of course you should argue that the dropdown shouldn't even be there then.